### PR TITLE
Fetch API

### PR DESCRIPTION
**Pull request by mohametalmeari**
- [ ] Send and receive data from an API.
- [ ] Make JavaScript code asynchronous.

### CODE REVIEW

**Review by 1GORDON — COMMENTED** [review]
##
## Hi @mohametalmeari ✋

Good job so far!

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

### This project will be marked as Invalid and won't affect the number of your weekly reviews.

- [ ] Kindly remove the Live demo section from your readme file since this project is not deployed. This is a general requirement for all projects.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — COMMENTED** [review]
Hi @mohametalmeari , 
image

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ ❌ 

- [ ] You have submitted a project with an unprofessional `readme`. Please remove all guiding texts and edit the document to fit your project. Kindly also follow the previous reviewers suggestions. 👍🏾 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/modules/game-id.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      .then((response) => response.json())
+      .then((json) => {
+        const myGameID = JSON.stringify(json).match(/Game with ID: (\w+)/)[1];
+        localStorage.setItem('myGameID', myGameID);
+        resolve(myGameID);
+      });
+  } else {
+    resolve(localStorage.getItem('myGameID'));
+  }
+});
```
Comment (wayungi) [inline src/modules/game-id.js:21]: - [ ] As suggested by the previous reviewer, please use the **async & await** pattern in your API calls 😄

File: src/modules/util.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const RefreshList = (url) => {
+  const getData = () => new Promise((resolve) => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((json) => resolve(json));
+  });
```
Comment (wayungi) [inline src/modules/util.js:6]: - [ ] Please use the async & await pattern here too. This is a project requirement 😄

File: src/modules/util.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const msg = document.getElementById('msg');
+  const sendData = () => new Promise((resolve) => {
+    const myname = document.getElementById('name');
+    const myscore = document.getElementById('score');
+    if (myscore.value < 0 || myscore.value > 100) {
+      msg.innerHTML = 'Score must be between 0 and 100.';
+    } else {
+      fetch(url, {
+        method: 'POST',
+        body: JSON.stringify({
```
Comment (wayungi) [inline src/modules/util.js:32]: - [ ] Just as I pointed out above about the requirement to use async-await, I'll suggest you update this part of your code too 😄

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @mohametalmeari , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] Nice application interface 👍🏾 
- [ ] Linter checks passed 👍🏾 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ OPTIONAL]  `>`  in the readme marks a line as a guiding text. All such lines should be edited and have the `>` removed or better still delete the whole line

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @mohametalmeari , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] Nice application interface 👍🏾 
- [ ] Linter checks passed 👍🏾 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ OPTIONAL]  `>`  in the readme marks a line as a guiding text. All such lines should be edited and have the `>` removed or better still delete the whole line

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — APPROVED** [review]
Hi @mohametalmeari , **we got it this time** 🥳 

image

Great work, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Remember: Every finish line 🏁  is the beginning of a new race 🐎** 

## Highlights
- [ ] Good use of async & await 👏🏾 
- [ ] All requested changes implemented 👏🏾 
- [ ] All milestone features have been implemented 👏🏾 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — APPROVED** [review]
Hi @mohametalmeari , **we got it this time** 🥳 

image

Great work, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Remember: Every finish line 🏁  is the beginning of a new race 🐎** 

## Highlights
- [ ] Good use of async & await 👏🏾 
- [ ] All requested changes implemented 👏🏾 
- [ ] All milestone features have been implemented 👏🏾 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**